### PR TITLE
Add International Edition

### PR DESCRIPTION
**Pull request by robertberry-zz**
CC @gklopper 

This is probably buggy as hell. Putting it out here so @michaelwmcnamara can help me start testing it.

Still needs design love for the 'beta' bit, too.

screen shot 2015-04-08 at 15 41 19

screen shot 2015-04-08 at 15 41 29

### CODE REVIEW

File: common/app/common/Edition.scala [hunk 1]
```diff
@@ -90,13 +119,30 @@ object Editionalise {
...
   private val variants = Seq("control", "international")
 
-  def apply(request: RequestHeader): Option[InternationalEdition] = request.headers.get("X-GU-International")
-    .filter(variants.contains)
-    .map(InternationalEdition(_))
+  def apply(request: RequestHeader): Option[InternationalEdition] = {
+    if (Switches.InternationalEditionSwitch.isSwitchedOn) {
+      val fromHeader = request.headers.get("X-GU-International").map(_.toLowerCase)
+
+      val fromCookie = request.cookies.get("GU_EDITION").map(_.value.toLowerCase)
```
Comment (gklopper) [inline common/app/common/Edition.scala]: Cookies don't work in PROD (they get wiped at the router).

We probably want to add the `X-GU-Edition` header here...

File: facia/test/FaciaControllerTest.scala [hunk 2]
```diff
@@ -101,4 +102,26 @@ import services.ConfigAgent
...
+    val redirectToUk = FaciaController.renderFront("")(control)
+    header("Location", redirectToUk).head should endWith ("/uk")
+
+    val international = FakeRequest("GET", "/").withHeaders("X-GU-Edition" -> "INTL", "X-GU-International" -> "international")
+    val redirectToInternational = FaciaController.renderFront("")(international)
+    header("Location", redirectToInternational).head should endWith ("/international")
+
+    val ukOptInRequest = FakeRequest("GET", "/").withHeaders("X-GU-Edition" -> "UK", "X-GU-International" -> "international")
+    val redirect = FaciaController.renderFront("")(international)
+    header("Location", redirect).head should endWith ("/uk")
```
Comment (gklopper) [inline facia/test/FaciaControllerTest.scala]: I added this breaking test which I think proves there is something not quite right with the `isInternational` logic.

Anyone who chooses a different edition should get that edition in preference to the international edition.

File: common/app/common/Edition.scala [hunk 3]
```diff
@@ -90,13 +119,33 @@ object Editionalise {
...
 }
 
-case class InternationalEdition(variant: String)
+case class InternationalEdition(variant: String) {
+  def isControl = variant == "control"
+
+  def isInternational = !isControl
+}
 
 object InternationalEdition {
```
Comment (gklopper) [inline common/app/common/Edition.scala]: I think this needs a comprehensive set of Unit tests

File: common/app/common/Edition.scala (hunk 1 above)
Comment (robertberry) [inline common/app/common/Edition.scala]: Cool, I'll add that.

**Comment by robertberry** [comment]
@gklopper Your test should now pass

**Comment by gklopper** [comment]
:+1: - I'm happy with this for now. As long as we do not switch it on in PROD yet it is good to go.

**Comment by robertberry** [comment]
Think it's all ready

screen shot 2015-04-09 at 12 35 49

screen shot 2015-04-09 at 12 35 59

screen shot 2015-04-09 at 12 36 07

**Pull request merged**